recipe/linux/activate.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# Set `CUDA_HOME` in an activation script.
+
+# Backup environment variables (only if the variables are set)
+if [[ ! -z "${CUDA_HOME+x}" ]]
+then
+  export CUDA_HOME_CONDA_NVCC_BACKUP="${CUDA_HOME:-}"
+fi
+
+if [[ ! -z "${CUDA_PATH+x}" ]]
+then
+  export CUDA_PATH_CONDA_NVCC_BACKUP="${CUDA_PATH:-}"
+fi
+
+if [[ ! -z "${CFLAGS+x}" ]]
+then
+  export CFLAGS_CONDA_NVCC_BACKUP="${CFLAGS:-}"
+fi
+
+if [[ ! -z "${CPPFLAGS+x}" ]]
+then
+  export CPPFLAGS_CONDA_NVCC_BACKUP="${CPPFLAGS:-}"
+fi
+
+if [[ ! -z "${CXXFLAGS+x}" ]]
+then
+  export CXXFLAGS_CONDA_NVCC_BACKUP="${CXXFLAGS:-}"
+fi
+
+if [[ ! -z "${CMAKE_ARGS+x}" ]]
+then
+  export CMAKE_ARGS_CONDA_NVCC_BACKUP="${CMAKE_ARGS:-}"
+fi
+
+# Default to using $(cuda-gdb) to specify $(CUDA_HOME).
+if [[ -z "${CUDA_HOME+x}" ]]
+then
+    CUDA_GDB_EXECUTABLE=$(which cuda-gdb || exit 0)
+    if [[ -n "$CUDA_GDB_EXECUTABLE" ]]
+    then
+        CUDA_HOME=$(dirname $(dirname $CUDA_GDB_EXECUTABLE))
+    else
+        echo "Cannot determine CUDA_HOME: cuda-gdb not in PATH"
+        return 1
+    fi
+fi
+
+if [[ ! -d "${CUDA_HOME}" ]]
+then
+    echo "Directory specified in CUDA_HOME(=${CUDA_HOME}) doesn't exist"
+    return 1
+fi
+
+if [[ ! -f "${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
+then
+    echo "File ${CUDA_HOME}/lib64/stubs/libcuda.so doesn't exist"
+    return 1
+fi
+
+if [[ -z "$(${CUDA_HOME}/bin/nvcc --version | grep "Cuda compilation tools, release __PKG_VERSION__")" ]]
+then
+  if [[ "${CONDA_BUILD}" == "1" ]]
```

Comment:
Sorry to push back on this, but the single equals was here because that is what [POSIX shells do]( https://stackoverflow.com/a/20449556 ) (`==` is unique to Bash). Would suggest we keep the POSIX compliant version

```suggestion
  if [[ "${CONDA_BUILD}" = "1" ]]
```

<br>
cc @jaimergp (for awareness)

Reply:
In that case shouldn't we use single brackets? Both work though, so no big deal.

Reply:
Good point. Fixed above in an edit.

On a larger point, it would be nice to have shell agnostic activation/deactivation scripts ( https://github.com/conda/conda/issues/6820 ), but that is outside of the scope here.

Reply:
Thanks for raising the discussion John. I wasn't as keen on the different shell conventions as I'd like to, so it's good that you caught it. If you could help we can try to make it POSIX complaint in this PR, but I am not knowledgeable enough (nor do I have the bandwidth) to do it myself 😅

Reply:
We could add `shellcheck` to the test requirements and start running it on these scripts to check compliance

That said, I wouldn't worry about doing this in this PR. We can save it for future work ( https://github.com/conda-forge/nvcc-feedstock/issues/92 )